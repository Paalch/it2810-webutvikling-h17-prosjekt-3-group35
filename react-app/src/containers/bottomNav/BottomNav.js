import React, {Component} from 'react';
import './bottomNav.css';
import {Row, Col} from 'react-bootstrap';


export class BottomNav extends React.Component {

  render() {
    return (
      <Col>
        <div className={'nav-container'}>
          <Row className={'item-container'}>
            <Col xs={4} id={'schedule'} onClick={this.props.openSchedule}>Schedule</Col>
            <Col xs={4} id={'todo'} onClick={this.props.openTodo}>2do</Col>
            <Col xs={4} id={'notes'} onClick={this.props.openNotes}>Notes</Col>
          </Row>
        </div>
      </Col>
    );
  }

}

export default BottomNav;