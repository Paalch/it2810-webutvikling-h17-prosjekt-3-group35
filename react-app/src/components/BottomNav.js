import React, { Component } from 'react';
import '../styles/bottomNav.css'
import {Row, Col} from 'react-bootstrap';

class BottomNav extends Component {
    render () {
        return (
            <div className={'nav-container'}>
                <Row className={'item-container'}>
                    <Col xs={4} lgHidden mdHidden smHidden id={'schedule'} className={'nav-items'}>Schedule</Col>
                    <Col xs={4} lgHidden mdHidden smHidden id={'todo'} className={'nav-items'}>2do</Col>
                    <Col xs={4} lgHidden mdHidden smHidden id={'notes'} className={'nav-items'}>Notes</Col>
                </Row>
            </div>
        );
    }
}

export default BottomNav;