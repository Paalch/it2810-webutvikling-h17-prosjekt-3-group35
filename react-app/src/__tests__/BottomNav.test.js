import React from 'react'
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-15'
import BottomNav from '../containers/bottomNav/bottomNav'
import Enzyme, { shallow } from 'enzyme'

Enzyme.configure({ adapter: new Adapter() })

describe('Checks that the navbar renders and contains what it must', () => {
    it('Checks that the navbar renders', () => {
        const rendered = renderer.create(
            <BottomNav />
        );
        expect(rendered.toJSON()).toMatchSnapshot()
    });

    it('Checks that all parts of the app is in the navbar', () => {
        const wrapper = shallow(<BottomNav />);
        expect(wrapper.find('#todo').exists()).toBeTruthy();
        expect(wrapper.find('#notes').exists()).toBeTruthy();
        expect(wrapper.find('#schedule').exists()).toBeTruthy();
    });
});
