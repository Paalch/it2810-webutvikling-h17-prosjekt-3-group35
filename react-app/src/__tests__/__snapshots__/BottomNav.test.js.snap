// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checks that the navbar renders and contains what it must Checks that the navbar renders 1`] = `
<div
  className="nav-container row"
>
  <div
    className="nav-item col-xs-4"
    id="schedule"
    onClick={undefined}
  >
    Schedule
  </div>
  <div
    className="nav-item col-xs-4"
    id="todo"
    onClick={undefined}
  >
    2do
  </div>
  <div
    className="nav-item col-xs-4"
    id="notes"
    onClick={undefined}
  >
    Notes
  </div>
</div>
`;
